### PR TITLE
Enable pylint check C0209

### PR DESCRIPTION
**Pull request by Shrews**
Enables pylint check C0209 (consider-using-f-string) and changes all non-f-string formats.

Also, change yamllint config to ignore files specified in `.gitignore` so local runs of the `linters` tox environment are more likely to succeed.

### CODE REVIEW

File: src/ansible_runner/config/runner.py [hunk 1]
```diff
@@ -270,35 +269,35 @@ def generate_ansible_command(self):
...
                 extra_vars_list = []
                 for k in self.extra_vars:
-                    extra_vars_list.append("\"{}\":{}".format(k, json.dumps(self.extra_vars[k])))
+                    extra_vars_list.append(f"\"{k}\":{json.dumps(self.extra_vars[k])}")
 
                 exec_list.extend(
                     [
                         '-e',
-                        '{%s}' % ','.join(extra_vars_list)
+                        f'{{{",".join(extra_vars_list)}}}'
```
Comment (Akasurde) [inline src/ansible_runner/config/runner.py:283]:
```suggestion
                        ",".join(extra_vars_list)
```

**Review by Akasurde — APPROVED** [review]

File: src/ansible_runner/config/runner.py (hunk 1 above)
Comment (Shrews) [reply src/ansible_runner/config/runner.py:283]: Your suggestion isn't the same, though. The intent is to pass the extra vars as inline JSON so the surrounding parens are required here.

Comment (Shrews) [reply src/ansible_runner/config/runner.py:283]: The triple `{` and `}` do look weird, but the first two are how you send a literal paren character in a f-string. The third is for the interpolated value.

Comment (Akasurde) [reply src/ansible_runner/config/runner.py:283]: TIL

**Pull request merged**